### ISSUE TITLE
Turn off test_core_python.py by default

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#4868]
`pytest_generate_tests` in `src/tests/test_core_python.py` is quite slow and causes a noticeable delay in test collection. I think we should exclude it from the tests by default.

**Comment by ryanking13** [comment]
I agree that core Python test suites are quite robust and rarely break, except for when updating the Python version of the emscripten version.

Maybe we can run it only in the main branch after merge and disable it in the PR? Another option would be periodically running full test suites.

**Comment by hoodmane** [comment]
Oh, I was thinking we would continue to run it in CI always but disable it by default locally. At least to start with, it might also make sense to only run it on the main branch.

**Comment by ryanking13** [comment]
Oh, I see. I didn't feel the test collection to be that slow, but if it's checked properly in CI, I don't think it's a big deal to skip it.
